pmlc/conversion/pxa_to_affine/pxa_to_affine.cc

```diff
@@ -57,6 +57,38 @@ struct LoweringBase : public OpConversionPattern<OpType> {
   LogicalResult match(Operation *op) const override { return mlir::success(); }
 };
 
+// This pattern removes affine.parallel ops with no induction variables
+struct AffineParallelRank0Remover
+    : public mlir::OpRewritePattern<AffineParallelOp> {
+  using mlir::OpRewritePattern<AffineParallelOp>::OpRewritePattern;
+
+  // Default benefit of 2 since this generally needs to run before
+  // AffineParallelOp conversion
+  explicit AffineParallelRank0Remover(MLIRContext *ctx,
```

Comment:
I think rather than use benefit (which does work but eventually becomes complex to reason about since you need to order things carefully and it's an integer), we should modify the AffineParallelOpConversion to have a match which fails when the number of dims is == 0 (since in fact it *doesnt* work in that case).  Then only one of the two patterns will be applicable at any given time.

Reply:
Oh, clever, let’s add a clear comment for this since it’s not entirely obvious. But seems reasonable.

Reply:
I'll put together another PR to revise w/ this comment